flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/assigners/MySqlSnapshotSplitAssigner.java

```diff
@@ -291,7 +291,7 @@ public Optional<MySqlSplit> getNext() {
             } else if (!remainingTables.isEmpty()) {
                 try {
                     // wait for the asynchronous split to complete
-                    lock.wait();
+                    lock.wait(100);
```

Comment:
Please use a longer timeout(e.g. 4s) and use a constant to maintain this fixed value.